### PR TITLE
[groovyscripting] Add Groovy scripting support

### PR DESCRIPTION
**Pull request by wborn**
This makes it really easy to start using [Groovy](https://groovy-lang.org/) 3.x scripts with openHAB!

---

```shell
openhab> log:set info jsr223.groovy
openhab> log:tail
20:05:00.470 [INFO ] [jsr223.groovy                        ] - Hello World!
```

### CODE REVIEW

**Review by cpmeister — APPROVED** [review]
LGTM

File: bundles/org.openhab.automation.groovyscripting/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<features name="org.openhab.automation.groovyscripting-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.4.0">
+	<repository>mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-core/${ohc.version}/xml/features</repository>
+
+	<feature name="openhab-automation-groovyscripting" description="Groovy Scripting" version="${project.version}">
+		<feature>openhab-runtime-base</feature>
+		<feature dependency="true">openhab-core-automation-module-script-rulesupport</feature>
```
Comment (wborn) [inline bundles/org.openhab.automation.groovyscripting/src/main/feature/feature.xml:7]: This dependency could be removed when https://github.com/openhab/openhab-core/pull/1725 gets merged.

**Comment by wborn** [comment]
There's more than one maintainer review required to get this merged. It's not a lot of code. ;-)

**Review by kaikreuzer — APPROVED** [review]
lgtm and even Jenkins is now happy ;-)

**Comment by kaikreuzer** [comment]
@wborn One thing that came to my mind last night is that we will also need to have a README file for those new add-ons, so that it can be included in the documentation (under [this link](https://github.com/openhab/openhab-core/pull/1722/files#diff-5456a839b084e7b88d042fdcd6fb60498372b93c4eb01a5471585ac7a6bb318bR112)).

Would you want to try to come up with a decent template/structure on how such a README should look like for automation add-ons?

**Comment by wborn** [comment]
Yes I agree that the current README.md should have some more substance. :-)

Some ideas I already had for it are:
* a link to the offical scripting language documentation (already there)
* a simple stand alone example to explain how to create a rule using a file programmatically
* a simple stand alone example to explain how to create a rule using the UI
* a link to [openhab-scripters](https://openhab-scripters.github.io/openhab-helper-libraries/index.html) for available helper libraries and additional docs
* a link to the common [JSR223 Scripting](https://www.openhab.org/docs/configuration/jsr223.html) documentation
* a link to the [JavaDocs](https://next.openhab.org/javadoc/latest/index.html)
* explain settings when applicable (we could make it configurable if for instance (embedded) helpers libraries are used).

In case of Jython the [existing docs](https://www.openhab.org/docs/configuration/jsr223-jython.html#jsr223-jython-scripting) should probably also be moved to the add-on. That is because you can no longer get any additional JSR223 engine working by dropping in some JARs in `/runtime/lib/ext` with Java 11 (see [community post](https://community.openhab.org/t/beta-testers-wanted-jython-addon-w-helper-libraries-requires-oh-2-5-x/86633/239?u=wborn)).

We probably also want to apply the same template to the existing [Nashorn docs](https://www.openhab.org/docs/configuration/jsr223-js.html#jsr223-javascript-nashorn) which now works out of the box.

Perhaps @openhab-5iver, @jpg0 and @Confectrician also have some additional ideas on this subject?
If possible we should not duplicate documentation to make it more easily maintainable.

**Comment by kaikreuzer** [comment]
> Yes I agree that the current README.md should have some more substance. :-)

Ah, there actually is already one - it was so short that I overlooked it 😆 .

- a simple stand alone example to explain how to create a rule using a file programmatically
- a link to the common JSR223 Scripting documentation

I think these would be the most important ones as a first step to get the README filled with some content.

> In case of Jython the existing docs should probably also be moved to the add-on.

Yes, fully agree.

**Pull request merged**